game-core/src/test/java/games/strategy/triplea/odds/calculator/DummyPlayerTest.java

```diff
@@ -0,0 +1,189 @@
+      when(unitType.getAttachment(any())).thenReturn(unitAttachment);
+      when(unitAttachment.getIsAir()).thenReturn(!unit.equals(unitPool.get(0)));
+      final var gameData = mock(GameData.class);
+      when(unit.getData()).thenReturn(gameData);
+      when(gameData.getResourceList()).thenReturn(mock(ResourceList.class));
+      final var playerId = mock(PlayerId.class);
+      when(unit.getOwner()).thenReturn(playerId);
+    }
+
+    final DummyPlayer player = new DummyPlayer(null, false, "", null, true, 0, 0, false);
+    final CasualtyDetails details =
+        player.selectCasualties(
+            unitPool,
+            null,
+            0,
+            null,
+            null,
+            null,
+            null,
+            null,
+            false,
+            null,
+            defaultCasualties,
+            null,
+            null,
+            false);
+    assertThat(details.getDamaged(), is(damaged));
+    assertThat(details.getKilled(), is(List.of(unitPool.get(1), unitPool.get(2))));
+    assertThat(details.getAutoCalculated(), is(false));
+  }
+
+  @Test
+  void testSelectCasualties_noLand_noLossOrder() {
+    final DummyPlayer player = new DummyPlayer(null, false, "", null, false, 0, 0, false);
+    final CasualtyDetails details =
+        player.selectCasualties(
+            unitPool,
+            null,
+            0,
+            null,
+            null,
+            null,
+            null,
+            null,
+            false,
+            null,
+            defaultCasualties,
+            null,
+            null,
+            false);
+    assertThat(details.getDamaged(), is(damaged));
+    assertThat(details.getKilled(), is(killed));
+    assertThat(details.getAutoCalculated(), is(false));
+  }
+
+  @Test
+  void testSelectCasualties_oneLand() {
+    for (final var unit : unitPool) {
+      final var unitType = mock(UnitType.class);
+      when(unit.getType()).thenReturn(unitType);
+      final var unitAttachment = mock(UnitAttachment.class);
+      when(unitType.getAttachment(any())).thenReturn(unitAttachment);
+      when(unitAttachment.getIsAir()).thenReturn(!unit.equals(unitPool.get(0)));
+      final var gameData = mock(GameData.class);
+      when(unit.getData()).thenReturn(gameData);
+      when(gameData.getResourceList()).thenReturn(mock(ResourceList.class));
+      final var playerId = mock(PlayerId.class);
+      when(unit.getOwner()).thenReturn(playerId);
+    }
+
+    final DummyPlayer player = new DummyPlayer(null, false, "", orderOfLosses, true, 0, 0, false);
+    final CasualtyDetails details =
+        player.selectCasualties(
+            unitPool,
+            null,
+            0,
+            null,
+            null,
+            null,
+            null,
+            null,
+            false,
+            null,
+            defaultCasualties,
+            null,
+            null,
+            false);
+    assertThat(details.getDamaged(), is(damaged));
```

Comment:
`damaged` is an expected list? If so, renaming the variable to `expectedDamage` could help make that more obvious (obvious is good, it means faster comprehension).

Assertion comments I think would go a long way in helping to keep these tests maintainable to describe why we expect the assertions to hold. Will that reasoning remain obvious to you after a few months/years?

Reply:
Well this list should contain damaged units?
I'd assume that the headed player implementation uses this list in order to indicate which 2-hit battleships or something already took a hit, or should take a hit as part of the result of the method.
Just guesses though